packages/sanity/src/core/studio/components/navbar/search/components/recentSearches/item/RecentSearchItem.tsx

```diff
@@ -0,0 +1,148 @@
+import {ClockIcon, CloseIcon} from '@sanity/icons'
+import {
+  Box,
+  Button,
+  Flex,
+  rem,
+  ResponsiveMarginProps,
+  ResponsivePaddingProps,
+  Text,
+} from '@sanity/ui'
+import React, {MouseEvent, useCallback} from 'react'
+import styled from 'styled-components'
+import {useSearchState} from '../../../contexts/search/useSearchState'
+import type {RecentSearch} from '../../../datastores/recentSearches'
+import {DocumentTypesPill} from '../../common/DocumentTypesPill'
+import {FilterPill} from '../../common/FilterPill'
+
+export interface RecentSearchesProps extends ResponsiveMarginProps, ResponsivePaddingProps {
+  index: number
+  maxVisibleTypePillChars?: number
+  showFiltersOnClick?: boolean
+  value: RecentSearch
+}
+
+const DEFAULT_COMBINED_TYPE_COUNT = 40
+
+const RecentSearchItemButton = styled(Button)`
+  border-radius: ${({theme}) => rem(theme.sanity.radius[2])};
+  cursor: default;
+  width: 100%;
+`
+
+const SearchItemPillsBox = styled(Box)`
+  flex-shrink: 3;
+`
+
+const SearchItemQueryBox = styled(Box)`
+  flex-shrink: 2;
+`
+
+const CloseButtonDiv = styled.div`
+  opacity: 0.8;
+  visibility: hidden;
+
+  @media (hover: hover) {
+    ${RecentSearchItemButton}:hover & {
+      visibility: visible;
+    }
+    &:hover {
+      opacity: 0.4;
+    }
+  }
+`
+
+export function RecentSearchItem({
+  index,
+  maxVisibleTypePillChars = DEFAULT_COMBINED_TYPE_COUNT,
+  showFiltersOnClick,
+  value,
+  ...rest
+}: RecentSearchesProps) {
+  const {dispatch, recentSearchesStore} = useSearchState()
+
+  // Determine how many characters are left to render type pills
+  const availableCharacters = maxVisibleTypePillChars - value.query.length
+
+  const handleClick = useCallback(() => {
+    const hasFilters = value.filters && value.filters.length
+    const hasTypes = value.types.length
+
+    // Optionally show filters panel if search terms or filters are present
+    if (showFiltersOnClick && (hasFilters || hasTypes)) {
+      dispatch({type: 'FILTERS_VISIBLE_SET', visible: true})
+    }
+    dispatch({type: 'TERMS_SET', filters: value?.filters, terms: value})
+
+    // Add to Local Storage
+    if (recentSearchesStore) {
+      const updatedRecentSearches = recentSearchesStore?.addSearch(value, value?.filters)
+      dispatch({recentSearches: updatedRecentSearches, type: 'RECENT_SEARCHES_SET'})
+    }
+  }, [dispatch, recentSearchesStore, showFiltersOnClick, value])
+
+  const handleDelete = useCallback(
+    (event: MouseEvent) => {
+      event.stopPropagation()
+      // Remove from Local Storage
+      if (recentSearchesStore) {
+        const updatedRecentSearches = recentSearchesStore?.removeSearchAtIndex(index)
+        dispatch({recentSearches: updatedRecentSearches, type: 'RECENT_SEARCHES_SET'})
+      }
+    },
+    [dispatch, index, recentSearchesStore]
+  )
+
+  return (
+    <Box {...rest}>
+      <RecentSearchItemButton
+        data-command-list-item
```

Comment:
Do we need this data attribute here? It seems to make the pressed styling disappear since `CommandListContainer` targets that attribute in its styling and overrides the background color.